src/canvas/SelectableCanvas.ts

```diff
@@ -490,6 +494,12 @@ export class SelectableCanvas<
   protected declare _isCurrentlyDrawing: boolean;
   declare freeDrawingBrush?: BaseBrush;
   declare _activeObject?: FabricObject;
+  protected readonly _activeSelection: ActiveSelection;
```

Comment:
is this necessary? or is just avoiding us to create an ActiveSelection when needed?
I would rather create on need rather than right away.
Is it a matter of code complexity or there is more?

Reply:
It is meant for controlling the stack order of active selection
https://github.com/fabricjs/fabric.js/issues/6840
https://github.com/fabricjs/fabric.js/issues/4529

Reply:
You linked to feature requests for this feature of which i m not against fixing.
I guess i m asking, why the _activeSelection needs to be always created and added to the canvas? it does not seem we are leveraging that anywhere, we could create it when is needed? or something is blocking us?

Reply:
We are not leveraging the fact that the activeSelection is prebuilt.
We save maybe a check to create it but then we add a check to empty it on selection cleared.
Where is the benefit?

Reply:
I m also thinking, how do i keep the click order if i don't want the stack order? i can't see where preserve object stacking actually makes the decision?

Reply:
The benefits:
```ts
const a = canvas.getActiveSelection()
a.on('object:added', ({target}) =>a.sendObjectToBack(target))

```
But this might be achieved with selection events

The other benefit is the ability to set the instance, meaning the dev can subclass active selection and use it w/o turning core upside down

Reply:
It is not mandatory to initialize it in constructor. I can do it when needed

Reply:
> I m also thinking, how do i keep the click order if i don't want the stack order? i can't see where preserve object stacking actually makes the decision?

can you explain more? I can understand this in a few ways

Reply:
Never did active selection have a way to control how objects are stacked in it

Reply:
No it didn't it captured them in click order.
Now it seems that it capture in stack order.
But why one is wrong and one is good?

Reply:
> No it didn't, it captured them in click order.

This is good. This is why we should hold a ref to active selection. So the dev can decide and run custom logic **OR** we could have an option for this on active selection: `selectionStrategy: 'click' | 'preserve-stacking'`.
At the time it made more sense to me to change it. It isn't better.
I would like to add this option.
What do you think?

for reference:
https://github.com/fabricjs/fabric.js/pull/8665/files#diff-47131ad6e0b321305307e39da0359c85513edd9b084e1fe1ef4941eab9e3f262L1536

https://github.com/fabricjs/fabric.js/pull/8665/files#diff-47131ad6e0b321305307e39da0359c85513edd9b084e1fe1ef4941eab9e3f262R1516-R1526

Reply:
> is this necessary? or is just avoiding us to create an ActiveSelection when needed? I would rather create on need rather than right away. Is it a matter of code complexity or there is more?

> It is not mandatory to initialize it in constructor. I can do it when needed

I have changed my mind. It is necessary. How would you add an event listener before there is a ref?

Reply:
`necessary` is out of topic here. Wasn't here, before and you could handle the selection just fine.
During the selection:created event you can attach events to the new selection.

But also i want to stress out that is unrelated. We are adding support for multi selection with objects located in different groups, and adding events on active selections, or offering a fixed reference to it, is a completely parallel issue that has been added here because it seemed a good idea while doing it.

There are reasons why in my opinion is not a good idea, one is that it make indivisible canvas and group and active selection, and that is worse than having to add events.
It is not wrong to have a fixed ref, but is not good to create it with canvas creation right away.
Having a ref allow us to not have imports to create the active selection, and that helps in dividing the code.